app/src/test/java/com/github/onlynotesswent/utils/NotesToFlashcardTest.kt

```diff
@@ -0,0 +1,203 @@
+import org.mockito.Captor
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.Mockito.anyString
+import org.mockito.Mockito.doAnswer
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.`when`
+import org.mockito.MockitoAnnotations
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class NotesToFlashcardTest {
+  // Function to allow null arguments for Mockito when needed
+  private fun <T> any(): T = Mockito.any()
+
+  // Helper function to capture arguments in Mockito tests, bypassing Kotlin's null-safety checks
+  private fun <T> capture(argumentCaptor: ArgumentCaptor<T>): T = argumentCaptor.capture()
+
+  private val jsonResponse =
+      """
+    {
+        "id": "chatcmpl-ASApJxQH9B975zpiNEukghJeWKMsd",
+        "object": "chat.completion",
+        "created": 1731278165,
+        "model": "gpt-3.5-turbo-0125",
+        "choices": [
+            {
+                "index": 0,
+                "message": {
+                    "role": "assistant",
+                    "content": "[{\"question\": \"What is cryptocurrency?\",\"answer\": \"Cryptocurrency is a digital payment system...\"},{\"question\": \"How does cryptocurrency work?\",\"answer\": \"Cryptocurrencies run on a distributed public ledger...\"},{\"question\": \"Cryptocurrency examples\",\"answer\": \"There are thousands of cryptocurrencies. Some of the best known include: Bitcoin, Ethereum...\"}]"
+                },
+                "logprobs": null,
+                "finish_reason": "stop"
+            }
+        ]
+    }
+    """
+          .trimIndent()
+
+  private val testNote =
+      Note(
+          id = "1",
+          title = "title",
+          content = "content",
+          date = Timestamp.now(),
+          visibility = Note.Visibility.DEFAULT,
+          userId = "1",
+          folderId = "1",
+          noteClass = Note.Class("CS-100", "Sample Class", 2024, "path"),
+          image = Bitmap.createBitmap(1, 1, Bitmap.Config.ARGB_8888))
+
+  private lateinit var notesToFlashcard: NotesToFlashcard
+
+  // Mock dependencies
+  @Mock private lateinit var mockFlashcardRepository: FlashcardRepository
+
+  @Mock private lateinit var mockFlashcardViewModel: FlashcardViewModel
+
+  @Mock private lateinit var mockFileViewModel: FileViewModel
+
+  @Mock private lateinit var mockOpenAI: OpenAI
+
+  // Argument captor for Flashcard objects
+  @Captor private lateinit var flashcardCaptor: ArgumentCaptor<Flashcard>
+
+  @Before
+  fun setup() {
+    MockitoAnnotations.openMocks(this)
+
+    // Initialize FirebaseApp with Robolectric context
+    FirebaseApp.initializeApp(org.robolectric.RuntimeEnvironment.getApplication())
+
+    // Mock FlashcardRepository and set up FlashcardViewModel with it
+    mockFlashcardRepository = mock(FlashcardRepository::class.java)
+    mockFlashcardViewModel = FlashcardViewModel(mockFlashcardRepository)
```

Comment:
technically you don't need to call the view model mock, and the repository is already mocked with the openMocks on line 89. not that important